src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -133,40 +149,50 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     if (!queueEntry.IsAbandoned && !queueEntry.IsCompleted)
                         await queueEntry.AbandonAsync().AnyContext();
                 }
+            }, new OnMessageOptions {
+                AutoComplete = false
             });
         }
 
         public override async Task<IQueueEntry<T>> DequeueAsync(TimeSpan? timeout = null) {
+            await EnsureQueueCreatedAsync().AnyContext();
```

Comment:
We could use the same lock in `DeleteQueueAsync()` as `EnsureQueueCreatedAsync()` to prevent this?

I started this PR with it returning the queue client but moved away to have more code in the base class.